llvm/test/SYCLLowerIR/printf_generic_as.ll

```diff
@@ -0,0 +1,102 @@
+;; This tests replacement of string literal address space for __spirv_ocl_printf
+;; at the regular O2 optimization level.
+
+;; Compiled with the following command (custom build of SYCL Clang with
+;; SYCLMutatePrintfAddrspacePass turned off):
+;; clang++ -fsycl -fsycl-device-only experimental-printf.cpp -S -D__SYCL_USE_NON_VARIADIC_SPIRV_OCL_PRINTF__
+;;
+;; // experimental-printf.cpp
+;; #include <CL/sycl.hpp>
+;; using namespace sycl;
+;; int main() {
+;;   queue q;
+;;   q.submit([&](handler &cgh) {
+;;     cgh.single_task([=]() {
+;;       ext::oneapi::experimental::printf("String No. %f\n", 1.0f);
+;;       const char *IntFormatString = "String No. %i\n";
+;;       ext::oneapi::experimental::printf(IntFormatString, 2);
+;;       ext::oneapi::experimental::printf(IntFormatString, 3);
+;;     });
+;;   });
+;;   return 0;
+;; }
```

Comment:
I suggest to put this SYCL code under `Inputs` subdirectory as a separate file. It shortens the test file, but still makes your test maintainable, because the source file is available.

`Inputs` is usually ignored by LIT (If I'm not mistaken, that happens by default, at least in existing LLVM LIT config files for tests)